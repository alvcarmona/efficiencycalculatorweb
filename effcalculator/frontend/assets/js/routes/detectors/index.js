/**
 contains a Route. Again, this goes down the path of breaking things down into as many small components as possible.
 I would recommend this (as oppoesd to having a route hierarchy written in XML) as it allows you to instantiate
 routes only as you need them. This also means that your route file will only including other files in relative
 sub-directories, which feels nice and de-coupled.
 Your route file can then also be the guardian of the data in your store when you go to the route, by making use of the
 onEnter and onLeave methods. In here you can dispatch fetch actions that ensure that your components have the data
 they need. This is really useful when you have deep nested routes.
 */

/**
 * Created by alvarocbasanez on 12/07/17.
 */
import React, {Component} from 'react';
import DetectorListContainer from './containers/DetectorListContainer';
import DetectorDetailContainer from './containers/DetectorDetailContainer';
import DetectorEditContainer from './containers/DetectorEditContainer';
import DetectorEfficiencyContainer from './containers/DetectorEfficiencyContainer';
import {Route, Switch} from 'react-router-dom'
import {connect} from 'react-redux';
import {fetchData, setCurrentDetector} from '../../modules/actions/index';
import {bindActionCreators} from 'redux';

function mapStateToProps(state) {
    return {data: state.data}
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({fetchData, setCurrentDetector}, dispatch)
}

class DetectorEfficiencyCalculator extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        this.props.fetchData()
    }

    render() {
        return (
            <div className="DetectorEfficiencyCalculator">
                <Switch>
                    <Route exact path='/Detectors' component={DetectorListContainer}/>
                    <Route exact path='/Detectors/:number' component={DetectorDetailContainer}/>
                    <Route path='/Detectors/:number/edit' component={DetectorEditContainer}/>
                    <Route path='/Detectors/:number/efficiency' component={DetectorEfficiencyContainer}/>
                </Switch>
            </div>
        );
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(DetectorEfficiencyCalculator)
