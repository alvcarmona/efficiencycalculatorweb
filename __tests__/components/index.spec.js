import React from 'react';
import {shallow} from 'enzyme';
import {DetectorEfficiencyCalculator} from '../../effcalculator/frontend/assets/js/routes/detectors/index';
import NavBarContainer from '../../effcalculator/frontend/assets/js/routes/NavbarContainer';

describe('DetectorEfficiencyCalculator', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DetectorEfficiencyCalculator fetchData={jest.fn()}/>);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a <DetectorEfficiencyCalculator />', () => {
    expect(wrapper.find('.DetectorEfficiencyCalculator').length).toEqual(1);
  });
});