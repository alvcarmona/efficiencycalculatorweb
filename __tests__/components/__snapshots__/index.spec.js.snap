// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetectorEfficiencyCalculator should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetectorEfficiencyCalculator
    fetchData={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/Detectors"
        />
      </Switch>,
      "className": "DetectorEfficiencyCalculator",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Route
          component={[Function]}
          exact={true}
          path="/Detectors"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/Detectors",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/Detectors"
          />
        </Switch>,
        "className": "DetectorEfficiencyCalculator",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Route
            component={[Function]}
            exact={true}
            path="/Detectors"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/Detectors",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
